patches/tModLoader/Terraria/Social/Steam/WorkshopSocialModule.cs.patch

```diff
@@ -9,3 +9,12 @@
  	{
  		private WorkshopHelper.UGCBased.Downloader _downloader;
  		private WorkshopHelper.UGCBased.PublishedItemsFinder _publishedItems;
+@@ -24,7 +_,7 @@
+ 			_publisherInstances = new List<WorkshopHelper.UGCBased.APublisherInstance>();
+ 			base.ProgressReporter = new WorkshopProgressReporter(_publisherInstances);
+ 			base.SupportedTags = new SupportedWorkshopTags();
+-			_contentBaseFolder = Main.SavePath + Path.DirectorySeparatorChar + "Workshop";
++			_contentBaseFolder = Directory.GetParent(Main.SavePath).ToString() + Path.DirectorySeparatorChar + "Workshop";
```

Comment:
prefer `Path.Combine` 
It looks like this is only used for `GetTemporaryFolderPath` so there's no reason for it to be common to `Terraria`, or across instances?

Reply:
On the note of "GetTemporaryFolderPath", the name is inaccurate - that folder is NOT temporary.
It is the folder where workshop files are stored prior to upload and persists across pushed updates for delta comparison by Steam.

I do notice now that the fact that folder is localized to a computer may create issues with the preview system in that if you publish on a new computer it will not have the history files of the mod. Something to look in to.